include/arch/arc/v2/linker.ld

```diff
@@ -224,6 +224,16 @@ SECTIONS {
 	__kernel_ram_end = .;
 	__kernel_ram_size = __kernel_ram_end - __kernel_ram_start;
 
+#ifdef __MWDT_LINKER_CMD__
+/* mwdt requires _fstack, _estack which will be used in _stkchk.
+ * _stkchk is inserted by mwdt automatically, if _fstack, _estack is not
+ * set correctly, the  brk_s instruction will be called
+ * here, we use a trick to deceive the compiler.
```

Comment:
Isn't there an option to disable this? I guess not, if you propose this workaround. Can you check with the MetaWare team, or ask for such an option?

Reply:
The only option is to rebuild MWDT libraries with stack check disabled. (well we can rebuild MWDT libraries with buildlib from Zephyr but I really don't like such option :)

I'm going to discuss it with MWDT team so we can probably ship yet another prebuilt library version specially targeted for Zephyr. There are several other things we want to change - I'm collecting the list of changes now.

But anyway that require time and for this Zephyr release we wave to use workarounds.